### PR TITLE
LibWeb+LibGfx: Add an initial implementation of SVG gradients (just `<linearGradient>` for now)

### PR DESCRIPTION
**Pull request by MacDue**
This adds an initial version of SVG gradients (where to simplify things a little, I've gone a little ad-hoc in places).

This PR adds support for the `<stop>` and `<linearGradient>` elements and for the gradients supports:
- The `gradientUnits` attribute 
- The `gradientTransform` attribute 
- Following `xlink:hrefs` for gradients that inherit stops/attributes 
- Other cool things, trust me :^) 

This is enough to now correctly render (as far as I can tell from a thorough eyeballing) all the gradients on the Discord login page:

image

as well as all my amazing test cases (not all shown):
image

### CODE REVIEW

File: Userland/Libraries/LibWeb/SVG/TagNames.h [hunk 1]
```diff
@@ -30,7 +30,9 @@ namespace Web::SVG::TagNames {
     __ENUMERATE_SVG_TAG(desc)          \
     __ENUMERATE_SVG_TAG(foreignObject) \
     __ENUMERATE_SVG_TAG(script)        \
-    __ENUMERATE_SVG_TAG(title)
+    __ENUMERATE_SVG_TAG(title)         \
+    __ENUMERATE_SVG_TAG(stop)          \
```
Comment (nico) [inline Userland/Libraries/LibWeb/SVG/TagNames.h:34]: This used to be alphabetized :redlinus:

File: Userland/Libraries/LibWeb/DOM/ElementFactory.cpp [hunk 2]
```diff
@@ -431,6 +432,8 @@ static WebIDL::ExceptionOr<JS::GCPtr<SVG::SVGElement>> create_svg_element(JS::Re
         return MUST_OR_THROW_OOM(realm.heap().allocate<SVG::SVGForeignObjectElement>(realm, document, move(qualified_name)));
     if (local_name == SVG::TagNames::line)
         return MUST_OR_THROW_OOM(realm.heap().allocate<SVG::SVGLineElement>(realm, document, move(qualified_name)));
+    if (local_name == SVG::TagNames::stop)
+        return MUST_OR_THROW_OOM(realm.heap().allocate<SVG::SVGStopElement>(realm, document, move(qualified_name)));
     if (local_name == SVG::TagNames::path)
```
Comment (nico) [inline Userland/Libraries/LibWeb/DOM/ElementFactory.cpp:440]: These ifs are also (mostly) alphabetized.

(Not sure why some do `==` and some do `equals_ignoring_case`)

File: Userland/Libraries/LibWeb/SVG/AttributeNames.h [hunk 3]
```diff
@@ -65,6 +65,7 @@ namespace Web::SVG::AttributeNames {
     E(specularConstant)             \
     E(specularExponent)             \
     E(spreadMethod)                 \
+    E(offset)                       \
```
Comment (nico) [inline Userland/Libraries/LibWeb/SVG/AttributeNames.h:68]: same

File: Userland/Libraries/LibWeb/idl_files.cmake [hunk 4]
```diff
@@ -205,6 +205,7 @@ libweb_js_bindings(SVG/SVGPolygonElement)
 libweb_js_bindings(SVG/SVGPolylineElement)
 libweb_js_bindings(SVG/SVGRectElement)
 libweb_js_bindings(SVG/SVGSVGElement)
+libweb_js_bindings(SVG/SVGStopElement)
```
Comment (nico) [inline Userland/Libraries/LibWeb/idl_files.cmake:210]: :greenlinus:

File: Userland/Libraries/LibWeb/idl_files.cmake [hunk 5]
```diff
@@ -195,6 +195,7 @@ libweb_js_bindings(SVG/SVGDefsElement)
 libweb_js_bindings(SVG/SVGElement)
 libweb_js_bindings(SVG/SVGGeometryElement)
 libweb_js_bindings(SVG/SVGGraphicsElement)
+libweb_js_bindings(SVG/SVGGradientElement)
```
Comment (nico) [inline Userland/Libraries/LibWeb/idl_files.cmake:198]: Almost! `d` < `p` though!

File: Userland/Libraries/LibWeb/DOM/ElementFactory.cpp (hunk 2 above)
Comment (MacDue) [reply Userland/Libraries/LibWeb/DOM/ElementFactory.cpp:440]: I think SVG tag names are case sensitive, while HTML for HTML it does not matter

Comment (MacDue) [reply Userland/Libraries/LibWeb/DOM/ElementFactory.cpp:440]: Not sure about the ones like `clipPath` though :thinking:

File: Userland/Libraries/LibWeb/idl_files.cmake (hunk 5 above)
Comment (MacDue) [reply Userland/Libraries/LibWeb/idl_files.cmake:197]: :sweat_smile:

**Pull request merged**